.github/workflows/check-milestone.yml

```diff
@@ -47,34 +46,41 @@ jobs:
               process.exit(0);
             }
 
-            const issueId = getLinkedIssueId(pull);
+            // the github API doesn't expose linked issues, so we have to parse the body ourselves
+            const prDescription = pull.data.body;
+            const linkedIssues = getLinkedIssues(prDescription);
 
-            if (issueId) {
-              console.log('linked issue found', issueId);
+            if (Array.isArray(linkedIssues) && linkedIssues.length > 0) {
+              console.log('linked issue(s) found');
 
-              const issue = await github.rest.issues.get({
-                owner,
-                repo,
-                issue_number: issueId,
-              });
+              linkedIssues.forEach(async (issueId, index) => {
+                const issue = await github.rest.issues.get({
+                  owner,
+                  repo,
+                  issue_number: issueId,
+                });
 
-              const milestone = getMilestone(issue);
+                const milestone = getMilestone(issue);
 
-              if (milestone) {
-                console.log("Linked issue has milestone", milestone.title);
-                process.exit(0);
-              }
-            }
+                if (milestone) {
+                  console.log(`Linked issue #${issueId} has milestone`, milestone.title);
+                  index === linkedIssues.length -1 && process.exit(0);
+                } else {
+                    console.log("No milestone found");
 
-            console.log("No milestone found");
+                    const author = pull.data.user.login;
+                    const guideLink = "https://www.notion.so/metabase/Metabase-Branching-Strategy-6eb577d5f61142aa960a626d6bbdfeb3?pvs=4#3dea255ffa3b4f74942a227844e889fa";
+                    const message = `@${author} Did you forget to add a milestone to the issue #${issueId} linked in this PR? _[When and where should I add a milestone?](${guideLink})_`;
 
-            const author = pull.data.user.login;
-            const guideLink = "https://www.notion.so/metabase/Metabase-Branching-Strategy-6eb577d5f61142aa960a626d6bbdfeb3?pvs=4#3dea255ffa3b4f74942a227844e889fa";
-            const message = `@${author} Did you forget to add a milestone to the issue for this PR? _[When and where should I add a milestone?](${guideLink})_`;
+                    await github.rest.issues.createComment({
+                      owner,
+                      repo,
+                      issue_number: pullNumber,
+                      body: message,
+                    });
 
-            await github.rest.issues.createComment({
-              owner,
-              repo,
-              issue_number: pullNumber,
-              body: message,
-            });
+                    // Exit as soon as we found an issue without a milestone and alerted the author.
+                    process.exit(0);
```

Comment:
@iethree as the alternative, we can let the script go through ALL linked issues and alert for each unlinked issue individually. WDYT?

Reply:
i think one message is sufficient, i like how you did it here 👍